autoscaler/autoscaling_groups.py

```diff
@@ -281,18 +303,25 @@ def time_out_spot_asgs(self, asgs):
             instance_type = asg.launch_config['InstanceType']
             instance_asg_map.setdefault(instance_type, []).append(asg)
 
-        now = datetime.datetime.utcnow()
+        now = datetime.datetime.now(pytz.utc)
         since = now - datetime.timedelta(seconds=self._SPOT_HISTORY_PERIOD)
 
         for region, instance_asg_map in region_instance_asg_map.items():
+            # Expire old history
+            history = [item for item in self._spot_price_history.get(region, []) if item['Timestamp'] > since]
+            if history:
+                newest_spot_price = max([item['Timestamp'] for item in history])
```

Comment:
nit: we don't need the `[]` inside the `max()` i think, to avoid creating a list